Applications/Utils/SimpleMeshCreation/createMeshElemPropertiesFromASCRaster.cpp

```diff
@@ -35,212 +36,158 @@
+        "Generates properties for mesh elements of an input mesh deploying a "
+        "ASC raster file",
+        ' ',
+        "0.1");
+
+    TCLAP::ValueArg<std::string> out_mesh_arg(
+        "o",
+        "out-mesh",
+        "the mesh is stored to a file of this name",
+        false,
+        "",
+        "filename for mesh output");
+    cmd.add(out_mesh_arg);
+
+    TCLAP::ValueArg<bool> refinement_raster_output_arg(
+        "",
+        "output-refined-raster",
+        "write refined raster to a new ASC file",
+        false,
+        false,
+        "0");
+    cmd.add(refinement_raster_output_arg);
 
     TCLAP::ValueArg<unsigned> refinement_arg(
-            "r",
-            "refine",
-            "refinement factor that raises the resolution of the raster data",
-            false,
-            1,
-            "factor (default = 1)");
-    cmd.add( refinement_arg );
-
-    TCLAP::ValueArg<std::string> mapping_arg("",
-                                             "mapping-name",
-                                             "file name of mapping",
-                                             true,
-                                             "",
-                                             "file name");
-    cmd.add( mapping_arg );
+        "r",
+        "refine",
+        "refinement factor that raises the resolution of the raster data",
+        false,
+        1,
+        "factor (default = 1)");
+    cmd.add(refinement_arg);
 
     TCLAP::ValueArg<std::string> raster_arg("",
                                             "raster-file",
                                             "the name of the ASC raster file",
                                             true,
                                             "",
                                             "file name");
-    cmd.add( raster_arg );
+    cmd.add(raster_arg);
+
+    TCLAP::ValueArg<std::string> property_arg(
+        "p",
+        "property-name",
+        "the name of the property the values are stored for",
+        false,
+        "gwn",
+        "property name as string");
+    cmd.add(property_arg);
 
     TCLAP::ValueArg<std::string> mesh_arg("m",
                                           "mesh",
                                           "the mesh is read from this file",
                                           true,
                                           "test.msh",
                                           "file name");
-    cmd.add( mesh_arg );
+    cmd.add(mesh_arg);
+
+    std::vector<std::string> allowed_units{ "mm/a", "mm/month", "m/s" };
+    TCLAP::ValuesConstraint<std::string> allowed_units_constraints{
+        allowed_units};
+    TCLAP::ValueArg<std::string> unit_arg("u",
+                                          "input-unit",
+                                          "input unit of the data",
+                                          true,
+                                          "m/s",
```

Comment:
I'll move the conversion to different units into another tool that can be used separately. To answer your questions:
1. The conversion of values is sometimes necessary and a separate tool can be used in a workflow.
2. I restricted the values because in groundwater flow I had to deal with this unit combinations. I want to make it easy to handle for the user. Of course the the list of possible unit conversions will be more complete in the future.